src/core/exec-invoke.c

```diff
@@ -1077,27 +1077,92 @@ static int pam_close_session_and_delete_credentials(pam_handle_t *handle, int fl
         return r != PAM_SUCCESS ? r : s;
 }
 
+static int notify_pam_pid(
+                int notify_pam_pid_fd,
+                const PidRef *parent_pid) {
+
+        struct iovec iov;
+        struct msghdr msghdr = {
+                .msg_iov = &iov,
+                .msg_iovlen = 1,
+        };
+        int r;
+
+        assert(notify_pam_pid_fd >= 0);
+        assert(pidref_is_set(parent_pid));
+
+        _cleanup_(pidref_done) PidRef pam_pid = PIDREF_NULL;
+        r = pidref_set_self(&pam_pid);
+        if (r < 0)
+                return r;
+
+        char buf[DECIMAL_STR_MAX(pid_t) * 2 + STRLEN("\n")];
+        xsprintf(buf, PID_FMT"\n"PID_FMT, parent_pid->pid, pam_pid.pid);
+        iov = IOVEC_MAKE_STRING(buf);
+
+        bool send_pidfd = (parent_pid->fd > 0 && pam_pid.fd > 0);
```

Comment:
`>= 0` and `>= 0`. But `pidref_is_set` was already asserted above and `pidref_set_sefl` was called for the other pid, so this check is redundant?